File: js/therapy-v3/src/types/patient.ts
export interface Patient {
  id: number;
  full_name: string;
  phone: string;
  email: string | null;
  birth_date: string | null;
  notes: string | null;
  medical_history: string | null;
  created_at: string;
  updated_at: string;
}

export interface PatientDetail extends Patient {
  appointment_count: number;
}

export interface PatientCreateInput {
  full_name: string;
  phone: string;
  email?: string | null;
  birth_date?: string | null;
  notes?: string | null;
  medical_history?: string | null;
}

export interface PatientUpdateInput {
  full_name?: string;
  phone?: string;
  email?: string | null;
  birth_date?: string | null;
  notes?: string | null;
  medical_history?: string | null;
}

export interface PatientPaginatedResponse {
  total: number;
  items: Patient[];
  page: number;
  page_size: number;
}

File: python/therapy-api/src/therapy/patient/domain/model/patient.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class Patient:
    full_name: str
    phone: str
    id: int = 0
    email: str | None = None
    birth_date: datetime | None = None
    notes: str | None = None
    medical_history: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None
